addons/html_editor/static/src/components/history_dialog/history_dialog.scss

```diff
@@ -1,168 +1,276 @@
+        background: transparent var(--btn-close-bg) center / 1em auto no-repeat;
+        border: 0;
+        border-radius: $border-radius-sm;
+        opacity: var(--btn-close-opacity);
+        width: 2rem;
+
+        &.fa-expand::before {
+            content: "\f065";
         }
-    }
-    .history-view-inner {
-        padding: 8px 12px;
-        border: 1px solid #ddd;
-        border-top: 0;
-        overflow: auto;
-        .embedded-history-dialog-placeholder {
-            color: #444;
-            padding: 32px;
-            text-align: center;
-            font-size: 20px;
-            border : 1px solid #999;
-            border-radius: 4px;
-            margin: 8px 0;
-            background-image: linear-gradient(45deg, #d1d1d1 25%, #999 25%, #999 50%, #d1d1d1 50%, #d1d1d1 75%, #999 75%, #999);
-            background-size: 50px 50px;
-            text-shadow:
-                    -2px -2px 0 #d1d1d1,
-                    2px -2px 0 #d1d1d1,
-                    -2px 2px 0 #d1d1d1,
-                    2px 2px 0 #d1d1d1,
-                    -3px 0px 0 #d1d1d1,
-                    3px 0px 0 #d1d1d1,
-                    0px -3px 0 #d1d1d1,
-                    0px 3px 0 #d1d1d1,
-        }
-        .history-comparison-split {
-            position: relative;
+
+        &:hover {
+            opacity: 0.75;
+        }
+
+        &:focus {
+            outline: 0;
+            box-shadow: $btn-close-focus-shadow;
+            opacity: 1;
+        }
+
+        &:active {
+            border: none;
+            transform: none;
         }
+
+        &:disabled,
+        &.disabled {
+            pointer-events: none;
+            user-select: none;
+            opacity: 0.25;
+        }
+
+        @include media-breakpoint-down(md) {
+            display: none;
+        }
+    }
+
+    @include media-breakpoint-up(md) {
+        margin: $o-horizontal-padding * 5;
     }
-    .history-container {
-        --border-color: #ddd;
-        margin-left: 240px;
-        overflow: hidden;
-        .o_notebook_content {
-            padding: 10px 12px;
-            border: 1px solid var(--border-color);
-            border-top: 0;
-        }
-        .nav {
-            padding-left: 24px;
-        }
-        removed {
-            display: inline;
-            background-color: #f1afaf;
-            text-decoration: line-through;
+}
+
+.modal-fullscreen {
+    .o_expand_button.fa-expand:before {
+        content: "\f066";
+    }
+}
+
+.history-view-inner {
+    flex: 1;
+    min-height: 0;
+
+    .embedded-history-dialog-placeholder {
+        border: 2px dashed $border-color;
+
+        &:before {
+            content: '\f1b2';
+            font-family: FontAwesome;
+            font-size: 1.25rem;
             opacity: 0.5;
         }
-        added {
-            display: inline;
-            background-color: #c8f1af;
+    }
+}
+
+.history-container {
+    display: flex;
+    flex: 1;
+    min-width: 0;
```

Comment:
I'm sure we can achieve the same with utility classes using `flex-grow`, `d-flex`, and `d-none` for smaller breakpoints if needed.